tools/benchmark-extractor.py

```diff
@@ -0,0 +1,265 @@
+#! /usr/bin/python3
+
+"""
+Python3 script for parsing benchmark data from directories of log files, to produce a csv containing raw info.
+"""
+
+import os
+import sys
+import argparse
+import numpy as np
+import pandas as pd
+from collections import OrderedDict
+
+EXIT_FAILURE = 1
+EXIT_SUCCESS = 0
+
+LOG_EXT=".log"
+
+# Define command line arguments for this application
+def cli_args():
+    # Get arguments using argparse
+    parser = argparse.ArgumentParser(
+        description="Extract benchmark data from nemo2d stdout logs"
+        )
+    parser.add_argument(
+        "files",
+        type=str,
+        nargs="+",
+        help="paths to files or directories of log files to parse."
+        )
+
+    parser.add_argument(
+        "-b",
+        "--build",
+        type=str,
+        help="Hardcoded build flag. @temp.",
+        default="Unknown"
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=str,
+        help="location for output file"
+    )
+    parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="force overwriting of --output file."
+      )
+
+
+    args = parser.parse_args()
+    return args
+
+# Validate command line arguments for pre-empative abortion.
+def validate_args(args):
+    # Build the list of log files to be parsed.
+    valid_files = []
+    if args.files is not None and len(args.files) > 0:
```

Comment:
This is equivalent to `if args.files:` which is shorter and more Pythonic

Reply:
Changed in 9055076